transitions/core.py

```diff
@@ -359,16 +356,17 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
         self._transition_queue = deque()
         self.models = []
 
-        if initial is None:
-            self.add_states('initial')
+        if model is None and add_self:  # TODO(pbovbel) reconsider API for next major release
+            model = self
+
```

Comment:
Actually, add_self=False and no model is exactly the use case i'm interested in: definining a singleton, long-lived machine object, and then add/remove models (in-memory cache from db data) on the fly.

I also like the possibility of not instantiating a magic `'initial'` state, which is only possible if no models are specified and no `initial` arg is provided.

What kind of weird effects do your foresee?